www/js/control/general-settings.js

```diff
@@ -399,6 +399,23 @@ angular.module('emission.main.control',['emission.services',
             }
         });
     };
+    $scope.editUserProfile = function() {
+        CommHelper.getUser().then(function(profile){
+            const uuid = profile.user_id["$uuid"];
+            $state.go("root.main.enketosurvey", {
+                form_location: "json/user-profile_v1.json",
+                opts: JSON.stringify({
+                    session: {
+                        data_key: "manual/user_profile_survey",
```

Comment:
e.g. you could just populate the profile directly from here.

Reply:
We will be using user profile as external data for confirm survey in the yet-to-come future. So that's why I am implementing the service this way.

Reply:
But yes, I think it could work from either in the service or controller.

Reply:
For using the profile as external data, presumably you will want to retrieve the profile in the background on app startup and cache it locally instead of making remote calls while taking the survey. Again, since the profile is not trip specific and there is only one profile per app installation, it is much easier to work with.

Reply:
We are allowing the profile to be changed on-demand. So getting the most recent snapshot of the profile prior to launching the confirm survey is a must. I am told that the answer from the profile will be dynamically displayed on the confirm survey's multiple choice question(s)

Reply:
But you only allow users to modify the profile from the app, right? So you can cache the modified profile after any modifications occur as well.

Reply:
So perhaps moving back to BEMUserCache right?

Reply:
I was thinking more in terms of an in-memory cached value. Since you are already persisting this on the remote server and you can retrieve it on demand from the server, it seems like an in-memory variable would be the most efficient and easiest to use.